test/e2e/advancedSearch.spec.ts

```diff
@@ -0,0 +1,27 @@
+import { test, expect, Page } from "@playwright/test";
+import { AdvancedSearchPage } from "../pages/AdvancedSearchPage";
+import { SearchResultsPage } from "../pages/SearchResultsPage";
+
+let advancedSearchPage: AdvancedSearchPage;
+const searchedBookName: string = 'Robinson';
+const searchedAuthorName: string = 'Daniel Defoe';
+
+test.beforeEach(async({page}: {page: Page}) => {
+    advancedSearchPage = new AdvancedSearchPage({page});
+    await advancedSearchPage.navigateToAdvancedSearch();
+});
+
+test("advanced search heading is visible", async({page}: {page: Page}) => {
+    await expect(advancedSearchPage.advancedSearchHeading).toBeVisible();
+});
+
+test("advanced book search", async({page}: {page: Page}) => {
+    const searchResultsPage = await advancedSearchPage.search({
+        title: searchedBookName,
+        author: searchedAuthorName
+    });
+    await expect(page).toHaveURL(await searchResultsPage.getSearchResultsURL({title: searchedBookName, author: searchedAuthorName}));
```

Comment:
Just for better readability.
```suggestion
    const searchUrl = await searchResultsPage.getSearchResultsURL({title: searchedBookName, author: searchedAuthorName});
    await expect(page).toHaveURL(searchUrl);
```